otter/util/zk.py

```diff
@@ -170,3 +199,113 @@ def get_zk_dispatcher(kz_client):
             partial(perform_get_stat, kz_client),
         AcquireLock: perform_acquire_lock
     })
+
+
+class PollingLock(object):
+    """
+    Zookeeper lock recipe that polls the children on interval basis instead
+    of leaving a watch on previous child. It's supposed to replace
+    :obj:`kazoo.recipe.lock.Lock` and has same signature and _similar_ behavior
+    for ``acquire`` and ``release`` methods.
+    """
+
+    def __init__(self, dispatcher, path, identifier=""):
+        self.dispatcher = dispatcher
+        self.path = path
+        self.identifier = identifier
+        self._node = None
```

Comment:
> I could change the tests to drive through external interface but I believe that is being tested in test_acquire_* functions. I am trying to think if it is a bad idea to test internal state without driving it from public interface. Not sure.

This is another hint that Active Nothing would be a good idea here.  _Ideally_, you would never drive any behavior through a private interface.  If you find yourself needing to break encapsulation to talk to a private attribute, you should break the code into two layers, and test the behavior of the inner layer.  The main benefit of doing this is that it increases the value of the tests as _regression_ tests, because maintenance of the code's implementation details (for example replacing the `_node` attribute with some state-machine goop!) will not cause them to need to change.  The tension is that if you have to set up a _lot_ of state, it decreases their value as _unit_ tests, so sometimes you have to just accept some ugliness in the name of isolation.  In this case the state-setup seems pretty minimal and could be accomplished with a tiny function, so I think it should be done, as it would be a higher-level expression of intent: you want a `PollingLock` in a particular _state_, not one with an attribute set to a specific _value_.